views/view_package.dt

```diff
@@ -87,20 +93,40 @@ block body
 							|= depname
 						|= "- " ~ sp["description"].opt!string
 
-		hr
-		- if (readmeContents.length)
-			.repositoryReadme.markdown-body
-				- import vibe.textfilter.markdown : MarkdownFlags, MarkdownSettings, filterMarkdown;
-				- scope msettings = new MarkdownSettings;
-				- msettings.flags = MarkdownFlags.backtickCodeBlocks|MarkdownFlags.noInlineHtml|MarkdownFlags.tables;
-				- msettings.headingBaseLevel = 2;
-				- msettings.urlFilter = &urlFilter;
-				!= filterMarkdown(readmeContents, msettings)
+		.subtabsHeader
+			- if (readmeContents.length)
+				a.tab(class=activeTab == "info" ? "active" : "", href="#{req.rootDir}packages/#{normalizedPackageName}?tab=info") Info
+			- foreach (i, sample; sampleURLs)
+				a.tab(class=activeTab == "sample_"~i.to!string ? "active" : "", href="#{req.rootDir}packages/#{normalizedPackageName}?tab=sample_#{i}")= sample
+			a.tab.external(href="http://#{normalizedPackageName}.dpldocs.info/") Documentation
+		- bool renderedTabContent;
+		- if (activeTab == "info")
+			- if (readmeContents.length)
+				.repositoryReadme.markdown-body
+					- import vibe.textfilter.markdown : MarkdownFlags, MarkdownSettings, filterMarkdown;
+					- scope msettings = new MarkdownSettings;
+					- msettings.flags = MarkdownFlags.backtickCodeBlocks|MarkdownFlags.noInlineHtml|MarkdownFlags.tables;
+					- msettings.headingBaseLevel = 2;
+					- msettings.urlFilter = &urlFilter;
+					!= filterMarkdown(readmeContents, msettings)
+			- else
+				.repositoryReadme
+			- renderedTabContent = true;
+		- else
+			- foreach (i, sample; sampleURLs)
+				- if (activeTab == "sample_"~i.to!string)
+					- renderedTabContent = true;
+					include view_package.sample
```

Comment:
Maybe move this in a separate PR, s.t. documentation button can be rolled out before this feature is complete?

Reply:
if we go through with the review process quickly we can roll this out early too ;)